### PR TITLE
removed logging

### PR DESCRIPTION
**Pull request by fceller**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,13 @@
+v3.4.3 (2019-xx-xx)
+-------------------
+
+* removed logging of JWT token
+
+* added `--environment.delete` to allow to hide environment variables from
+  sub-processes or V8
```
Comment (Simran-B) [inline CHANGELOG]: `or` or `of` V8?

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+* hide senstive program options like ldap password from V8
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion
* hide sensitive program options like ldap password from V8
```

**Comment by fceller** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3274/

File: lib/V8/v8-utils.cpp [hunk 3]
```diff
@@ -307,8 +307,9 @@ static void JS_Options(v8::FunctionCallbackInfo<v8::Value> const& args) {
     TRI_V8_THROW_EXCEPTION_USAGE("options()");
   }
 
-  VPackBuilder builder =
-      ApplicationServer::server->options({"server.password"});
+  VPackBuilder builder = ApplicationServer::server->options(
+      {"server.password", "database.password", "ldap.bindpasswd",
```
Comment (jsteemann) [inline lib/V8/v8-utils.cpp]: Does an option `database.password` actually exist?

File: lib/ApplicationFeatures/EnvironmentFeature.cpp [hunk 4]
```diff
@@ -34,15 +36,26 @@
...
+void EnvironmentFeature::collectOptions(std::shared_ptr<options::ProgramOptions> options) {
+  options->addSection("environment", "Configure the environment");
+
+  options->addOption("--environment.delete",
+                     "delete the variables from the environment",
+                     new options::VectorParameter<options::StringParameter>(&_envDelete));
+}
+
 void EnvironmentFeature::prepare() {
+  for (auto env : _envDelete) {
```
Comment (jsteemann) [inline lib/ApplicationFeatures/EnvironmentFeature.cpp]:
```suggestion
  for (auto const& env : _envDelete) {
```

File: lib/ApplicationFeatures/EnvironmentFeature.cpp [hunk 5]
```diff
@@ -34,15 +36,26 @@
...
   setOptional(true);
   startsAfter("GreetingsPhase");
   startsAfter("MaxMapCount");
 }
 
+void EnvironmentFeature::collectOptions(std::shared_ptr<options::ProgramOptions> options) {
+  options->addSection("environment", "Configure the environment");
+
+  options->addOption("--environment.delete",
+                     "delete the variables from the environment",
```
Comment (jsteemann) [inline lib/ApplicationFeatures/EnvironmentFeature.cpp]:
```suggestion
                     "variables to remove from the environment, so they are not accessible by the ArangoDB process",
```

File: lib/V8/v8-utils.cpp (hunk 3 above)
Comment (fceller) [reply lib/V8/v8-utils.cpp]: Yes:

--database.password <string>                                initial password of root user (default: "")

**Comment by fceller** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3292/

**Pull request closed without merging**